tests/test_controller/test_action_normalization.py

```diff
@@ -0,0 +1,174 @@
+
+    controller = Controller(
+        ControllerConfig(action_normalizer="running_mean_std"),
+        task,
+        task_config,
+        optimizer,
+        optimizer_config,
+        rollout_backend="mujoco",
+    )
+
+    # Check that the normalizer is initialized as running_mean_std
+    assert isinstance(controller.action_normalizer, RunningMeanStdNormalizer)
+
+    # Test with some random actions
+    actions = np.random.randn(10, task.nu)
+    controller.action_normalizer.update(actions)
+
+    # Check that statistics were updated
+    assert controller.action_normalizer.count == 10
+    assert not np.allclose(controller.action_normalizer.mean, np.zeros(task.nu))
+    assert not np.allclose(controller.action_normalizer.std, np.ones(task.nu))
+
+    # Test normalization
+    normalized = controller.action_normalizer.normalize(actions)
+    denormalized = controller.action_normalizer.denormalize(normalized)
+
+    np.testing.assert_array_almost_equal(actions, denormalized)
+
+
+def test_normalizer_type_change() -> None:
+    """Test that normalizer is re-initialized when type changes."""
+    task_config = CylinderPushConfig()
+    task = CylinderPush()
+    optimizer_config = CrossEntropyMethodConfig()
+    optimizer = CrossEntropyMethod(optimizer_config, task.nu)
+
+    controller = Controller(
+        ControllerConfig(action_normalizer="identity"),
+        task,
+        task_config,
+        optimizer,
+        optimizer_config,
+        rollout_backend="mujoco",
+    )
+
+    # Initially should be identity normalizer
+    assert isinstance(controller.action_normalizer, IdentityNormalizer)
+
+    # Change the normalizer type in config
+    controller.controller_cfg.action_normalizer = "min_max"
+
+    # Update with some actions and rewards
+    actions = np.random.randn(10, task.nu)
+    rewards = np.random.randn(10)
+    controller.update_action_normalizer(actions, rewards)
+
+    # Should now be min_max normalizer
+    assert isinstance(controller.action_normalizer, MinMaxNormalizer)
+
+
+def test_normalizer_in_update_action_loop() -> None:
+    """Test that normalizers work in the update_action loop."""
+    task_config = CylinderPushConfig()
+    task = CylinderPush()
+    optimizer_config = CrossEntropyMethodConfig()
+    optimizer = CrossEntropyMethod(optimizer_config, task.nu)
+
+    # Test with different normalizer types
+    for normalizer_type in ["identity", "min_max", "running_mean_std"]:
+        controller = Controller(
+            ControllerConfig(action_normalizer=normalizer_type, max_opt_iters=2),
+            task,
+            task_config,
+            optimizer,
+            optimizer_config,
+            rollout_backend="mujoco",
+        )
+
+        curr_state = np.random.rand(controller.task.model.nq + controller.task.model.nv)
+        curr_time = 0.0
+
+        # This should run without error
+        controller.update_action(curr_state, curr_time)
```

Comment:
if you want to check that something does NOT raise an error, i like using this hack pattern for tests:
```python
from contextlib import nullcontext as does_not_raise

with does_not_raise():
    # some code that should run without issue
```